src/web3/contracts.ts

```diff
@@ -109,6 +121,36 @@ async function waitForContractKit(tries: number) {
   return contractKit
 }
 
+// This code assumes that the account for walletAddress already exists in the Keychain
+// which is a responsibility currently handled by KeychainWallet
+export function* getViemWallet(chain: Chain) {
+  if (viemWallets.has(chain)) {
+    return viemWallets.get(chain) as ViemWallet
+  }
+  const walletAddress = yield* select(walletAddressSelector)
+  if (!walletAddress) {
+    throw new Error('Wallet address not found')
+  }
+  const accounts = yield* call(listStoredAccounts)
+  const account = accounts.find((a) => a.address === walletAddress)
+  if (!account) {
+    throw new Error(`Account ${walletAddress} not found in Keychain`)
+  }
+  const password = yield* call(getPasswordSaga, walletAddress, false, true)
+  const privateKey = yield* call(getStoredPrivateKey, account, password)
+  if (!privateKey) {
+    throw new Error(`Private key not found for account ${walletAddress}`)
+  }
+  const wallet = getLockableViemWallet(
+    keychainLock,
+    chain,
+    normalizeAddressWith0x(privateKey) as Address
```

Comment:
Is the normalizeAddress required here? Seems like getStoredPrivateKey ensures this now with Jean's fix